lib/fs/storage.go

```diff
@@ -35,6 +35,20 @@ type Storage struct {
 	vol string
 }
 
+// List reads the path content
+func (s Storage) List(path string) ([]string, error) {
+	var err error
+
+	dir := fmt.Sprintf("%s%s", s.vol, strings.TrimPrefix(path, "/"))
```

Comment:
There's specific method to get full path to the directory, Instead of this you can do:

```go
loc, err := s.fullPath(path)
```